components/accept-financial-contributions/ContributionCategoryPicker.js

```diff
@@ -0,0 +1,202 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Flex, Box } from '@rebass/grid';
+import styled from 'styled-components';
+import { defineMessages, injectIntl } from 'react-intl';
+
+import { H1, P } from '../Text';
+import StyledButton from '../StyledButton';
+import Container from '../Container';
+import Link from '../Link';
+
+const Image = styled.img`
+  @media screen and (min-width: 52em) {
+    height: 256px;
+    width: 256px;
+  }
+  @media screen and (max-width: 40em) {
+    height: 192px;
+    width: 192px;
+  }
+  @media screen and (min-width: 40em) and (max-width: 52em) {
+    height: 208px;
+    width: 208px;
+  }
+`;
+
+class ContributionCategoryPicker extends React.Component {
+  static propTypes = {
+    query: PropTypes.object,
+    intl: PropTypes.object.isRequired,
+    onChange: PropTypes.func,
+  };
+
+  constructor(props) {
+    super(props);
+
+    this.messages = defineMessages({
+      header: {
+        id: 'acceptContributions.picker.header',
+        defaultMessage: 'Accept financial contributions',
+      },
+      subtitle: {
+        id: 'acceptContributions.picker.subtitle',
+        defaultMessage: 'Who will hold money on behalf of the Collective?',
+      },
+      myself: {
+        id: 'acceptContributions.picker.myself',
+        defaultMessage: 'Myself',
+      },
+      myselfInfo: {
+        id: 'acceptContributions.picker.myselfInfo',
+        defaultMessage: 'Use my personal bank account',
+      },
+      host: { id: 'acceptContributions.picker.host', defaultMessage: 'A fiscal host' },
+      hostInfo: {
+        id: 'acceptContributions.picker.hostInfo',
+        defaultMessage: 'Outsource fundholding to an entity who offers this service (More info)',
+      },
+      organization: { id: 'acceptContributions.picker.organization', defaultMessage: 'Our organization' },
+      organizationInfo: {
+        id: 'acceptContributions.picker.organizationInfo',
+        defaultMessage: 'Use my company or organization bank account',
+      },
+    });
+  }
+
+  handleChange(fieldname, value) {
+    this.props.onChange(fieldname, value);
+  }
+
+  render() {
+    const { intl, query } = this.props;
```

Comment:
Let's used `withRouter` and `this.props.router.query` instead of the current strategy.

Reply:
👍